go/controller/workflows.go

```diff
@@ -290,7 +290,11 @@ func (wf *Workflows) GetUserByRefreshTokenHash(
 	)
 	if errors.Is(err, pgx.ErrNoRows) {
 		logger.Error("could not find user by refresh token")
-		return sql.AuthUser{}, ErrInvalidPat //nolint:exhaustruct
+		e := ErrInvalidRefreshToken
+		if refreshTokenType == sql.RefreshTokenTypePAT {
+			e = ErrInvalidPat
+		}
+		return sql.AuthUser{}, e //nolint:exhaustruct
```

Comment:
nitpicking but this alternative is shorter and easier to read as it doesn't need to reference any variable (it also avoids an allocation)

```suggestion
		if refreshTokenType == sql.RefreshTokenTypePAT {
			return sql.AuthUser{}, ErrInvalidPat  //nolint:exhaustruct
		}
		return sql.AuthUser{}, ErrInvalidRefreshToken //nolint:exhaustruct
```